src/modules/Dashboard/features/Applet/Activities/Activities.tsx

```diff
@@ -1,59 +1,93 @@
+import { useTranslation } from 'react-i18next';
 import { useParams } from 'react-router-dom';
 
-import { getAppletActivitiesApi } from 'api';
+import { getAppletActivitiesApi, getAppletApi } from 'api';
+import { Spinner } from 'shared/components';
 import { useAsync } from 'shared/hooks';
 import {
   ActivitiesData,
   ActivityGrid,
   useActivityGrid,
 } from 'modules/Dashboard/components/ActivityGrid';
+import { FlowGrid } from 'modules/Dashboard/components/FlowGrid';
+import { Activity, ActivityFlow } from 'redux/modules';
+import { StyledFlexColumn } from 'shared/styles';
+
+import { ActivitiesSectionHeader } from './ActivitiesSectionHeader';
+import { ActivitiesToolbar } from './ActivitiesToolbar';
 
 export const Activities = () => {
   const { appletId } = useParams();
+  const { t } = useTranslation('app');
+  const [flows, setFlows] = useState<ActivityFlow[]>([]);
+  const [flowActivities, setFlowActivities] = useState<Activity[]>([]);
   const [activitiesData, setActivitiesData] = useState<ActivitiesData | null>(null);
-  const [isLoading, setIsLoading] = useState(true);
   const dataTestId = 'dashboard-applet-activities';
-
   const { formatRow, TakeNowModal } = useActivityGrid(dataTestId, activitiesData);
 
-  const { execute: getActivities } = useAsync(
+  const { execute: getActivities, isLoading: isLoadingActivities } = useAsync(
     getAppletActivitiesApi,
     (response) => {
       const activitiesDetails = response?.data.result.activitiesDetails;
 
       return setActivitiesData({ result: activitiesDetails, count: activitiesDetails.length });
     },
-    undefined,
-    () => setIsLoading(false),
   );
 
+  const { execute: getFlows, isLoading: isLoadingFlows } = useAsync(getAppletApi, (response) => {
+    if (response?.data?.result) {
+      setFlows(response.data.result.activityFlows ?? []);
+      setFlowActivities(response.data.result.activities ?? []);
+    }
+  });
+
+  const isLoading = isLoadingActivities || isLoadingFlows;
+
   useEffect(() => {
     if (!appletId) return;
-    getActivities({
-      params: {
-        appletId,
-      },
-    });
+    getFlows({ appletId });
+    getActivities({ params: { appletId } });
 
     return () => {
       setActivitiesData(null);
     };
-  }, [appletId, getActivities]);
+  }, [appletId, getActivities, getFlows]);
 
   const activities = useMemo(
     () => (activitiesData?.result ?? []).map((activity) => formatRow(activity)),
     [activitiesData, formatRow],
   );
 
   return (
-    <ActivityGrid
-      rows={activities}
-      TakeNowModal={TakeNowModal}
-      data-testid={dataTestId}
-      isLoading={isLoading}
-      order={'desc'}
-      orderBy={''}
-    />
+    <StyledFlexColumn sx={{ gap: 2.4, overflow: 'auto', p: 3.2 }}>
+      <ActivitiesToolbar appletId={appletId} data-testid={dataTestId} />
+
+      {isLoading ? (
+        <Spinner />
+      ) : (
+        <StyledFlexColumn sx={{ gap: 4.8 }}>
+          {flows?.length && (
```

Comment:
So that `0` doesn't render when it's empty:
```suggestion
          {!!flows?.length && (
```

Reply:
Good catch!